packages/build-utils/test/unit.get-project-paths.test.ts

```diff
@@ -0,0 +1,30 @@
+import path from 'path';
+import { getProjectPaths, ProjectPath } from '../src/get-project-paths';
+import { FixtureFilesystem } from './utils/fixture-filesystem';
+
+describe.each<[ProjectPath[], string, ProjectPath[]]>([
+  [
+    ['backend/app-three', 'frontend/app-one', 'frontend/app-two'],
+    '34-monorepo-no-workspaces',
+    [],
+  ],
+  [[], '35-no-monorepo', []],
+  [[], '32-monorepo-highly-nested', []],
+  [['backend/app-three'], '33-hybrid-monorepo', ['frontend']],
+])('`getProjectPaths()`', (paths, fixturePath, skipPaths) => {
+  const testName =
+    paths.length > 0
+      ? `should detect ${paths.join()} project${
+          paths.length > 1 ? 's' : ''
+        } for ${fixturePath}`
+      : `should not detect any path for ${fixturePath}`;
+
+  it(testName, async () => {
+    const fixture = path.join(__dirname, 'fixtures', fixturePath);
+    const fs = new FixtureFilesystem(fixture);
+
+    const actualPaths = await getProjectPaths({ fs, skipPaths });
+
+    expect(actualPaths).toEqual(expect.arrayContaining(paths));
```

Comment:
`arrayContaining` here is not a good proxy for what we're trying to do—it won't catch duplicates, which I hypothesize are present. I'd much prefer a `===` string comparison of `.sort().join("\u0000")` (since the null byte is invalid in path names on every system).

Reply:
(If this test passes after this change I *think* we're okay on my other comment about possible duplicate work.)